SQLAssignment.sql.sql

```diff
@@ -0,0 +1,1043 @@
+| 1001 | Peel    | London    |   12 |
+| 1002 | Serres  | SanJose   |   13 |
+| 1004 | Motika  | London    |   11 |
+| 1007 | Rifkin  | Barcelona |   15 |
+| 1003 | AxelRod | New York  |   10 |
+| 1008 | Fran    | London    |   25 |
++------+---------+-----------+------+
+2) List all customers with a rating of 100.
+ mysql> SELECT CNUM,CNAME,CITY,RATING,SNUM FROM customers WHERE RATING = 100;
++------+---------+--------+--------+------+
+| CNUM | CNAME   | CITY   | RATING | SNUM |
++------+---------+--------+--------+------+
+| 2001 | Hoffman | London |    100 | 1001 |
+| 2006 | Clemens | London |    100 | 1001 |
+| 2007 | Pereira | Rome   |    100 | 1004 |
++------+---------+--------+--------+------+
+3) Find the largest order taken by each salesperson on each date.
+
+mysql> SELECT SNUM ,ODATE,MAX(AMT) AS MAX_AMT FROM ORDERS GROUP BY ODATE,SNUM;
++------+------------+---------+
+| SNUM | ODATE      | MAX_AMT |
++------+------------+---------+
+| 1007 | 1990-10-03 | 1098.16 |
+| 1001 | 1990-10-03 |  767.19 |
+| 1004 | 1990-10-03 | 1900.10 |
+| 1002 | 1990-10-03 | 5160.45 |
+| 1003 | 1990-10-04 | 1713.23 |
+| 1002 | 1990-10-04 |   75.75 |
+| 1001 | 1990-10-05 | 4723.00 |
+| 1002 | 1990-10-06 | 1309.95 |
+| 1001 | 1990-10-06 | 9891.88 |
++------+------------+---------+
+4) Arrange the Order table by descending customer number.
+
+mysql> SELECT CNUM,ONUM,AMT,ODATE,SNUM FROM ORDERS ORDER BY CNUM DESC;
++------+------+---------+------------+------+
+| CNUM | ONUM | AMT     | ODATE      | SNUM |
++------+------+---------+------------+------+
+| 2008 | 3001 |   18.69 | 1990-10-03 | 1007 |
+| 2008 | 3006 | 1098.16 | 1990-10-03 | 1007 |
+| 2007 | 3002 | 1900.10 | 1990-10-03 | 1004 |
+| 2006 | 3008 | 4723.00 | 1990-10-05 | 1001 |
+| 2006 | 3011 | 9891.88 | 1990-10-06 | 1001 |
+| 2004 | 3007 |   75.75 | 1990-10-04 | 1002 |
+| 2004 | 3010 | 1309.95 | 1990-10-06 | 1002 |
+| 2003 | 3005 | 5160.45 | 1990-10-03 | 1002 |
+| 2002 | 3009 | 1713.23 | 1990-10-04 | 1003 |
+| 2001 | 3003 |  767.19 | 1990-10-03 | 1001 |
++------+------+---------+------------+------+
+5) Find which salespeople currently have orders in the order table.
+mysql> SELECT DISTINCT  SNAME FROM salespeople NATURAL JOIN Orders;
++---------+
+| SNAME   |
++---------+
+| Peel    |
+| Serres  |
+| AxelRod |
+| Motika  |
+| Rifkin  |
++---------+
+6) List names of all customers matched with the salespeople serving them.
+
+mysql> SELECT c.CNAME,s.SNAME FROM Customers c, Salespeople s WHERE  c.SNUM = s.SNUM;
++----------+---------+
+| CNAME    | SNAME   |
++----------+---------+
+| Hoffman  | Peel    |
+| Clemens  | Peel    |
+| Liu      | Serres  |
+| Grass    | Serres  |
+| Giovanni | AxelRod |
+| Pereira  | Motika  |
+| Cisneros | Rifkin  |
++----------+---------+
+7) Find the names and numbers of all salespeople who have more than one customer.
+
+mysql> SELECT c.SNUM,s.SNAME,COUNT(*) AS Customer_NO FROM Customers c, Salespeople s WHERE  c.SNUM = s.SNUM GROUP BY c.SNUM;
```

Comment:
This is a wrong or incomplete query. Please display only records for more than 1 customer.